### ISSUE TITLE
Support TV shows

### ISSUE THREAD

**Issue by TastyPi** [leepeuker/movary#186]
Apologies if they already are supported, I didn't see it mentioned in the README.

This is a feature request to support tracking TV show episode watches in addition to movies.

**Comment by kodxana** [comment]
+1 would use it too!

**Comment by leepeuker** [comment]
Hey, because this was asked at a few different places I have [updated the README](https://github.com/leepeuker/movary/commit/d6cdee877310f946f3b0d0067908e33595c867ea0) with an answer.

> Q: Will Movary support tv shows or other media types?
>
>A: No, there are currently no plans for this. The limited time available for developing Movary is used to provide the best possible experience for movie fans.

**Comment by iamkroot** [comment]
Would it be possible to keep this issue open with a "zero priority" label? This way interested folks can subscribe to the issue and get notified when the situation changes in the far future.

**Comment by leepeuker** [comment]
Hey @iamkroot, I would prefer letting this issue closed, I do not think that movary will ever support tv shows to be honest (it was planned and designed specifically for movies). 
Of course never say never, who knows what the future brings, but I do not like the idea to imply that it seems possible for tv shows to be added by lettings this issue open at the moment.

**Comment by JVT038** [comment]
Hey. I'm a PHP developer and I am willing to work on a PR that would add support for TV shows and music. 

I would probably fetch TV show info from TMDB (I think they have info about TV shows) and get music info from the free Spotify API. Besides that, I know Deezer, Musicbrainz and Genius have some pretty comprehensive publicly available APIs as well. 

Would you merge a PR that would add support for these things?

**Comment by leepeuker** [comment]
Hey @JVT038 , that is great, you are welcome to contribute and every PR will be merged, which adds value to this project without increasing the technical dept to much.

To your suggestions, my opinion is we should definitely forget about music and other media types for now, TV Show support alone will be enough work. If this is done we can think about supporting more media types.

So when we focus on TV right now, the first question would be how can we best integrate this in the existing web UI, without losing the possibility to ignore TV shows if a user has no interest in it :thinking: 

 As a starting point you could look into the `dashboard.html.twig` and find a way how we could add support for TV there.

**Comment by Jucgshu** [comment]
The README sounds like a hard 'No' atm. Perhaps a mention in it saying 'any contribution's welcome' could open up the discussion a little?

I must say that TV shows support is a must for me, although Movary seems really appealing.

**Comment by leepeuker** [comment]
Yeah good point @Jucgshu , I have [updated the readme}(https://github.com/leepeuker/movary/commit/da841e0354dbcc925541e95ca44ce1ff9e0e0f51) like you suggested. Additionally I have reopend this ticket for more visibility.

**Comment by Indian-Techie09** [comment]
@JVT038 instead of tmdb I think tvdb is better for TV shows

**Comment by JVT038** [comment]
The TVDB costs €12,- per year. [Source](https://thetvdb.com/subscribe)

I don't think it's desirable to ask users to pay €12,- per year for an API while there are free alternatives (such as TMDB)

**Comment by Indian-Techie09** [comment]
@JVT038 can you not use a v4 API instead of every user using their own api?

**Comment by JVT038** [comment]
I don't know for sure, but I have a pretty strong feeling that we're going to violate their API ToS if I create a free developer API key and distribute that to everyone using Movary.

**Comment by Indian-Techie09** [comment]
@JVT038 how about a caching proxy?

**Comment by JVT038** [comment]
What do you mean with a caching proxy?

**Comment by Indian-Techie09** [comment]
@JVT038 using a caching proxy for example squid we can cache data from tvdb than use that data for movary.

**Comment by JVT038** [comment]
Would each user be running their own proxy or would there be a centralized proxy for all users?

**Comment by Indian-Techie09** [comment]
I am thinking that each user setup their own proxy connected to each other like p2p

**Comment by JVT038** [comment]
I think building a whole p2p system with trackers, announcers and everything like that, would be too much effort if the TMDB can also just be used without the whole p2p system.
Besides that, other APIs like the Trakt.tv also provide a lot of information on TV shows, so I honestly don't see the point of making a very complicated proxy system with p2p while a simple API client can be made instead.

**Comment by JourneyOver** [comment]
Like the look of this application myself, but yea until tv shows are supported I sadly have to pass this app up even though I would like another place to have all my movies/tv shows backuped up to.

**Comment by DrMxrcy** [comment]
+1 On this!

**Comment by Drallas** [comment]
+1 for this feature..

**Comment by luiscachog** [comment]
+1 for this effort!

**Comment by starsoccer** [comment]
+1

**Comment by Klipperino** [comment]
tmdb.org also has tv shows, could we use this in Movary with the same API key we use for movies?
